### PR TITLE
Stops people from removing infinite power cell from combat defib

### PR DESCRIPTION
**Pull request by Tastyfish**
## About The Pull Request

The loaded combat defibrillator has an infinite power cell. You can remove it and have an infinite power cell to use as use please as a player.

This is probably bad.

Makes the combat defib's cell not removable.

## Why It's Good For The Game

Balance?

## Changelog

:cl:
fix: ERT and nuke ops players can no longer remove the power cell from a combat defibrillator.
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
Seems more like a fix. Such an item isnt really obtainable

File: code/game/objects/items/defib.dm [hunk 1]
```diff
@@ -22,6 +22,7 @@
 	var/on = FALSE //if the paddles are equipped (1) or on the defib (0)
 	var/safety = TRUE //if you can zap people with the defibs on harm mode
 	var/powered = FALSE //if there's a cell in the defib with enough power for a revive, blocks paddles from reviving otherwise
+	var/cell_removable = TRUE //Can the cell be removed?
```
[the remark this answers is not in the document]
Comment (Tastyfish) [reply code/game/objects/items/defib.dm]: Cleaned up all of the nearby comments to be autodocs

File: code/game/objects/items/defib.dm [hunk 2]
```diff
@@ -132,13 +135,16 @@
...
 
 /obj/item/defibrillator/screwdriver_act(mob/living/user, obj/item/tool)
-	if(cell)
-		cell.update_appearance()
-		cell.forceMove(get_turf(src))
-		cell = null
-		tool.play_tool_sound(src, 50)
-		to_chat(user, span_notice("You remove the cell from [src]."))
-		update_power()
+	if(!(cell && cell_removable))
```
[the remark this answers is not in the document]
Comment (Tastyfish) [reply code/game/objects/items/defib.dm]: Sure thing. Done.

File: code/game/objects/items/defib.dm [hunk 3]
```diff
@@ -132,13 +141,16 @@
...
-		to_chat(user, span_notice("You remove the cell from [src]."))
-		update_power()
+	if(!cell || !cell_removable)
+		return FALSE
+
+	cell.update_appearance()
+	cell.forceMove(get_turf(src))
+	cell = null
+	tool.play_tool_sound(src, 50)
+	to_chat(user, span_notice("You remove the cell from [src]."))
```
Comment (ShizCalev) [inline code/game/objects/items/defib.dm:151]: Move this up a hair so you can say" you remove [cell] from [src]"
Or just balloon alert "removed [cell]"

Comment (Tastyfish) [reply code/game/objects/items/defib.dm]: Sure, I love balloon alerts for feedback messages like this. Changed to a balloon alert.

**Comment by moocowswag** [comment]
Why not just make it a regular cell? Infinite power cell means people will realize "hey this dude is using a defib which uses energy" then tries to emp it to drain the cell only to find out it doesnt work...

**Comment by Tastyfish** [comment]
> Why not just make it a regular cell? Infinite power cell means people will realize "hey this dude is using a defib which uses energy" then tries to emp it to drain the cell only to find out it doesnt work...

That is more of an actual balance choice that I wouldn't be comfortable doing without maintainer approval.

**Pull request merged**